api/server/app.cc

```diff
@@ -456,34 +458,58 @@ static const char *SrcCgroup() {
     return NULL;
 }
 
+int setCgroups(std::string file1, std::string file2){        
+    std::ifstream ifs(file1);
+    std::ofstream ofs(file2);
+    if((ifs.is_open()) && (ofs.is_open())){
+	ofs << ifs.rdbuf();
+	return 0;
+    } else {
+        return -1;
+    }
+}
+
 static int InitCgroup(int multiplier) {
     const char *cgroupPath = SrcCgroup();
-
+    int check = 0;
+ 
     if (!cgroupPath)
         return -1;
-    std::string create_dir("mkdir " + std::string(cgroupPath) + "/butterfly");
-
-    BASH(create_dir) {
-        LOG_WARNING_("can't create butterfly cgroup, fail cmd '%s'",
-                     create_dir.c_str());
+    std::string directory = std::string(cgroupPath) + "/butterfly" ; 
+    if (mkdir((char*)directory.c_str(),S_IWGRP) < 0){
```

Comment:
mkdir take a const char * in parameter, so I don't think the cast is necessary.
Also, if you must cast, please use a C++ cast. (const_cast,static_cast,bit_cast,reinterpret_cast)